### PR TITLE
docs: make external-dns optional in echoserver walkthrough

### PR DESCRIPTION
**Pull request by prasadkatti**
### Description
This PR updates the `echoserver` walkthrough.

- makes it clear that use of `external-dns` is completely optional.
- adds a verification section that uses the load balancer DNS name in case `external-dns` is not used
- removes the `spec.rules[0].host` in the default manifest. With this in, the example does not work if external-dns is not used. The listener rule does not match because the Host Header does not match the ALB DNS name.
- removes instructions to deploy external-dns and instead links to the right page

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: docs/examples/echo_server.md [hunk 1]
```diff
@@ -5,8 +5,9 @@ In this walkthrough, you'll
 - Create a cluster with EKS
 - Deploy an aws-load-balancer-controller
 - Create deployments and ingress resources in the cluster
-- Use [external-dns](https://github.com/kubernetes-incubator/external-dns) to create a DNS record
-    - This assumes you have a route53 hosted zone available. Otherwise you can skip this, but you'll only be able to address the service from the ALB's DNS.
+- Verify access to the service
+- (Optional) Use [external-dns](https://github.com/kubernetes-incubator/external-dns) to create a DNS record pointing to the load balancer created by the aws-load-balancer-controller.
```
Comment (johngmyers) [inline docs/examples/echo_server.md:9]:
```suggestion
- (Optional) Use [external-dns](https://github.com/kubernetes-sigs/external-dns) to create a DNS record pointing to the load balancer created by the aws-load-balancer-controller.
```

File: docs/examples/echo_server.md [hunk 2]
```diff
@@ -143,8 +144,7 @@ In this walkthrough, you'll
                 alb.ingress.kubernetes.io/tags: Environment=dev,Team=test
         spec:
             rules:
-            - host: echoserver.example.com
-                http:
+            - http:
                 paths:
```
Comment (johngmyers) [inline docs/examples/echo_server.md:148]: The indentation is off here. Could you fix it?

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: docs/examples/echo_server.md (hunk 2 above)
Comment (prasadkatti) [reply docs/examples/echo_server.md:148]: It was actually off before. I fixed it here. Reference - https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#ingressrule-v1-networking-k8s-io

**Review by kishorj — APPROVED** [review]

**Pull request merged**